### PR TITLE
docs: add TOC example to template-only-components.md

### PR DESCRIPTION
**Pull request by lukasnys**
I've mostly been using the `TOC` syntax to define template-only components. However, I noticed there wasn't any official documentation for this.

### CODE REVIEW

File: docs/ember/template-only-components.md [hunk 1]
```diff
@@ -26,6 +26,23 @@ declare module '@glint/environment-ember-loose/registry' {
 
 Note that the runtime content of this module (effectively `export default templateOnlyComponent();`) is exactly what Ember generates at build time when creating a backing module for a template-only component.
 
+Additionally, you can create a backing module using the `TOC` type, which is a convenience alias for `templateOnlyComponent`.
+
+```typescript
```
Comment (NullVoxPopuli) [inline docs/ember/template-only-components.md:31]: adding an example file name here would be good, I think

**Comment by NullVoxPopuli** [comment]
thanks for submitting this PR!

File: docs/ember/template-only-components.md (hunk 1 above)
Comment (lukasnys) [reply docs/ember/template-only-components.md:33]: Makes sense to me! I added `app/components/shout.ts` as a filename

**Comment by lukasnys** [comment]
Closing in favor of #647

**Pull request closed without merging**